app/src/components/SearchBook.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { Autocomplete, CircularProgress, TextField } from '@mui/material';
+
+import { Book, CremonaClient } from 'cremona';
+
+interface SearchBookProps {
+  bookSearchHandler: Function;
+  disabled: boolean | undefined;
+}
+
+/**
+ * This component fixes an autocomplete with books and fires a handler when a bbok is autocompleted
+ * @param bookSearchHandler (String) => {....} Handles when a book is searched
+ * @returns SearchBook component
+ */
+export default function SearchBook({
+  bookSearchHandler,
+  disabled,
+}: SearchBookProps) {
+  const [open, setOpen] = React.useState(false);
+  const [options, setOptions] = React.useState<readonly Book[]>([]);
+  const loading = open && options.length === 0;
+
+  React.useEffect(() => {
+    let active = true;
+
+    if (!loading) {
+      return undefined;
+    }
+
+    (async () => {
+      const client = new CremonaClient();
```

Comment:
Using the raw CremonaClient may not be ideal since it reopens a websocket but I'm not sure whether to refactor this later or first add a cremona service and get that into this branch.